src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int CAPACITY_EMPTY_ELEMENT_DATA = 0;
```

Comment:
CAPACITY_EMPTY_ELEMENT_DATA seems to be a redundant constant as its value is 0, which is commonly understood as the default empty state for a collection. You can directly use 0 in your isEmpty() method.